_re2.cc

```diff
@@ -856,11 +904,25 @@ regexp_set_match(RegexpSetObject2* self, PyObject* text)
     return NULL;
   }
 
-  const char* raw_text = PyString_AsString(text);
+  const char* raw_text;
+  Py_ssize_t len_text;
+#if PY_MAJOR_VERSION >= 3
+  if (PyBytes_Check(text)) {
+    raw_text = PyBytes_AsString(text);
+    len_text = PyBytes_GET_SIZE(text);
+  } else if (PyUnicode_Check(text)) {
+    raw_text = PyUnicode_AsUTF8AndSize(text, &len_text);
+  } else {
+    PyErr_SetString(PyExc_TypeError, "expected str or bytes");
+    return NULL;
+  }
+#else
+  raw_text = PyString_AsString(text);
+  len_text = PyString_GET_SIZE(text);
```

Comment:
Need to check raw_text before invoking a macro that assumes text is a str.

Reply:
Fixed in next rev.